packages/tasit-action/src/NFT.test.js

```diff
@@ -43,9 +44,18 @@ describe("TasitAction.NFT", () => {
     expect(fullNFT.getAddress()).to.equal(fullNFTAddress);
     expect(fullNFT.name).to.exist;
     expect(fullNFT.symbol).to.exist;
-    expect(fullNFT.getProvider()).to.exist;
+    expect(fullNFT._getProvider()).to.exist;
+
+    provider = fullNFT._getProvider();
+
+    const ownerBalance = await fullNFT.balanceOf(owner.address);
```

Comment:
This `beforeEach` is getting a little big. Maybe consider stripping this out into a function that is either called in the `beforeEach` still or just the tests where it's pretty crucial

Reply:
I've just refactored, is it better now?

Reply:
I'll take a look!

Reply:
Yeah, I think it's better now. Let's aim for a rule of a beforeEach being no more than 20 lines

Reply:
I agree, `Contract.test.js` stills having large hooks (> 20) but I think that will be better after https://github.com/tasitlabs/TasitSDK/issues/149

Reply:
Yep! Resolving this